src/mono/mono.proj

```diff
@@ -464,11 +464,21 @@
     <!-- Linux options -->
     <ItemGroup Condition="'$(TargetsLinux)' == true">
       <_MonoCFLAGS Include="-Wl,--build-id=sha1" />
+      <_MonoCFLAGS Include="-Wno-int-conversion" />
       <_MonoCXXFLAGS Include="-Wl,--build-id=sha1" />
+      <_MonoCXXFLAGS Include="-Wno-int-conversion" />
     </ItemGroup>
     <ItemGroup Condition="'$(RealTargetOS)' == 'Linux'">
       <_MonoAOTCFLAGS Include="-Wl,--build-id=sha1" />
+      <_MonoAOTCFLAGS Include="-Wno-int-conversion" />
       <_MonoAOTCXXFLAGS Include="-Wl,--build-id=sha1" />
+      <_MonoAOTCXXFLAGS Include="-Wno-int-conversion" />
+    </ItemGroup>
+    <ItemGroup Condition="'$(TargetsLinuxMusl)' == 'true'">
+      <_MonoAOTCFLAGS Include="-lucontext" />
+      <_MonoAOTCXXFLAGS Include="-lucontext" />
+      <_MonoCFLAGS Include="-lucontext" />
+      <_MonoCXXFLAGS Include="-lucontext" />
     </ItemGroup>
```

Comment:
```suggestion
```

not needed, right?

Reply:
These were added as without it the libucontext library isn't pulled automatically on musl.

Reply:
You mentioned that (HP) libunwind was requiring libucontext on i386 and PowerPC64, and added that condition in `src/coreclr/pal/src/CMakeLists.txt`. Was that not enough, or is there something else requiring libucontext in mono? (for all architectures?)

Reply:
There's a mono specific requirement for `ucontext` on `s390x`:
```
  /usr/bin/s390x-alpine-linux-musl-ld: libmonosgen-2.0.a(exceptions-s390x.c.o): in function `mono_arch_get_restore_context':
  exceptions-s390x.c:(.text+0x14de): undefined reference to `setcontext'
  clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [mono/mini/CMakeFiles/mono-sgen.dir/build.make:194: mono/mini/mono-sgen] Error 1
  make[1]: *** [CMakeFiles/Makefile2:660: mono/mini/CMakeFiles/mono-sgen.dir/all] Error 2
  make: *** [Makefile:136: all] Error 2
/var/build/dotnet7/testing/dotnet7-stage0/src/dotnet-v7.0.100-rtm.22521.12/src/runtime/src/mono/mono.proj(551,5): error MSB3073: The command "TARGET_BUILD_ARCH=s390x PKG_CONFIG_PATH=/var/build/sysroot-s390x//usr/lib/s390x-linux-gnu/pkgconfig cmake --build . --target install --config Release --parallel 36" exited with code 2.
```
Can probably remove the CXXFLAGS component given that being a C file.

Reply:
Ok, `setcontext` is only used on s390x in mono, so if there is no requirement for other architectures, we can append this condition: `and '$(TargetArchitecture)' == 's390x'"`.

Reply:
Done